api/routes/auth0.ts

```diff
@@ -0,0 +1,45 @@
+import { FastifyPluginCallback } from 'fastify';
+
+declare module 'fastify' {
+  interface Session {
+    user: {
+      id: string;
+    };
+  }
+}
+
+export const auth0Routes: FastifyPluginCallback = (fastify, _options, done) => {
+  fastify.addHook(
+    'onRequest',
+    async (req, res) => await fastify.authenticate(req, res)
+  );
```

Comment:
```suggestion
  fastify.addHook('onRequest', fastify.authenticate);
```
`fastify.authenticate` is already an async function that takes two arguments, so there's no need to wrap it.